File: .skills/openclaw-skills/skills/bogorman/vector-robot/scripts/proxy-server.js
#!/usr/bin/env node
/**
 * OpenAI-compatible proxy for wire-pod → OpenClaw
 * Supports both streaming and non-streaming responses
 * 
 * Run: node proxy-server.js
 * Configure wire-pod: Custom provider, endpoint http://localhost:11435/v1
 */

const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = process.env.PROXY_PORT || 11435;
const REQUEST_FILE = path.join(__dirname, 'request.json');
const RESPONSE_FILE = path.join(__dirname, 'response.json');

// Clean up stale files
try { fs.unlinkSync(REQUEST_FILE); } catch {}
try { fs.unlinkSync(RESPONSE_FILE); } catch {}

const server = http.createServer(async (req, res) => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');

  if (req.method === 'OPTIONS') {
    res.writeHead(200);
    res.end();
    return;
  }

  if (req.url === '/health' || req.url === '/') {
    res.writeHead(200, { 'Content-Type': 'application/json' });
    res.end(JSON.stringify({ status: 'ok', service: 'vector-openclaw-proxy' }));
    return;
  }

  if (req.url === '/v1/models') {
    res.writeHead(200, { 'Content-Type': 'application/json' });
    res.end(JSON.stringify({
      data: [{ id: 'openclaw', object: 'model', owned_by: 'openclaw' }]
    }));
    return;
  }

  if (req.url === '/v1/chat/completions' && req.method === 'POST') {
    let body = '';
    req.on('data', chunk => body += chunk);
    req.on('end', async () => {
      try {
        const data = JSON.parse(body);
        const messages = data.messages || [];
        const isStreaming = data.stream === true;
        
        const userMessage = messages.filter(m => m.role === 'user').pop();
        const question = userMessage?.content || 'Hello';

        console.log(`[${new Date().toISOString()}] Question: "${question}" (stream: ${isStreaming})`);

        const request = { timestamp: Date.now(), question, raw: data };
        fs.writeFileSync(REQUEST_FILE, JSON.stringify(request, null, 2));

        // Wait for response
        const startTime = Date.now();
        const timeout = 25000;
        let response = null;

        while (Date.now() - startTime < timeout) {
          if (fs.existsSync(RESPONSE_FILE)) {
            try {
              const responseData = JSON.parse(fs.readFileSync(RESPONSE_FILE, 'utf8'));
              if (responseData.timestamp > request.timestamp - 1000) {
                response = responseData.answer;
                fs.unlinkSync(RESPONSE_FILE);
                break;
              }
            } catch (e) {}
          }
          await new Promise(r => setTimeout(r, 100));
        }

        if (!response) {
          response = "I didn't get a response in time. Please try again.";
          console.log(`[${new Date().toISOString()}] Timeout`);
        }

        console.log(`[${new Date().toISOString()}] Response: "${response}"`);

        if (isStreaming) {
          res.writeHead(200, {
            'Content-Type': 'text/event-stream',
            'Cache-Control': 'no-cache',
            'Connection': 'keep-alive'
          });

          const chunk = {
            id: 'chatcmpl-' + Date.now(),
            object: 'chat.completion.chunk',
            created: Math.floor(Date.now() / 1000),
            model: 'openclaw',
            choices: [{
              index: 0,
              delta: { role: 'assistant', content: response },
              finish_reason: null
            }]
          };
          res.write(`data: ${JSON.stringify(chunk)}\n\n`);

          const finishChunk = {
            id: 'chatcmpl-' + Date.now(),
            object: 'chat.completion.chunk',
            created: Math.floor(Date.now() / 1000),
            model: 'openclaw',
            choices: [{ index: 0, delta: {}, finish_reason: 'stop' }]
          };
          res.write(`data: ${JSON.stringify(finishChunk)}\n\n`);
          res.write('data: [DONE]\n\n');
          res.end();

        } else {
          const openaiResponse = {
            id: 'chatcmpl-' + Date.now(),
            object: 'chat.completion',
            created: Math.floor(Date.now() / 1000),
            model: 'openclaw',
            choices: [{
              index: 0,
              message: { role: 'assistant', content: response },
              finish_reason: 'stop'
            }],
            usage: { prompt_tokens: 0, completion_tokens: 0, total_tokens: 0 }
          };
          res.writeHead(200, { 'Content-Type': 'application/json' });
          res.end(JSON.stringify(openaiResponse));
        }

      } catch (e) {
        console.error('Error:', e);
        res.writeHead(500, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ error: e.message }));
      }
    });
    return;
  }

  res.writeHead(404, { 'Content-Type': 'application/json' });
  res.end(JSON.stringify({ error: 'Not found' }));
});

server.listen(PORT, () => {
  console.log(`Vector-OpenClaw proxy running on http://localhost:${PORT}`);
  console.log(`Configure wire-pod: Custom provider, endpoint http://localhost:${PORT}/v1`);
});

File: .skills/openclaw-skills/skills/r00tid/token-alert/scripts/proxy-server.py
#!/usr/bin/env python3
"""CORS Proxy Server for Token Alert Dashboard"""

from http.server import HTTPServer, SimpleHTTPRequestHandler
import urllib.request
import urllib.parse
import json
import sys

GATEWAY_URL = 'http://localhost:18789'
GATEWAY_TOKEN = 'd91a7a91e0d6bda8b6e3182467fda1f0bebd34c830263a4f'

class ProxyHandler(SimpleHTTPRequestHandler):
    def end_headers(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization')
        SimpleHTTPRequestHandler.end_headers(self)
    
    def do_OPTIONS(self):
        self.send_response(200)
        self.end_headers()
    
    def do_POST(self):
        # Proxy /api/* requests to Gateway
        if self.path.startswith('/api/'):
            gateway_path = self.path.replace('/api', '', 1)
            self.proxy_to_gateway(gateway_path)
        else:
            self.send_error(404, 'Not found')
    
    def proxy_to_gateway(self, path):
        try:
            # Read request body
            content_length = int(self.headers.get('Content-Length', 0))
            body = self.rfile.read(content_length) if content_length > 0 else b''
            
            # Forward to Gateway
            url = f'{GATEWAY_URL}{path}'
            headers = {
                'Authorization': f'Bearer {GATEWAY_TOKEN}',
                'Content-Type': 'application/json'
            }
            
            req = urllib.request.Request(url, data=body, headers=headers, method='POST')
            
            with urllib.request.urlopen(req) as response:
                response_data = response.read()
                
                # Send response
                self.send_response(response.status)
                self.send_header('Content-Type', 'application/json')
                self.end_headers()
                self.wfile.write(response_data)
                
        except urllib.error.HTTPError as e:
            self.send_response(e.code)
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(e.read())
            
        except Exception as e:
            self.send_error(500, f'Proxy error: {str(e)}')

if __name__ == '__main__':
    port = 8765
    server = HTTPServer(('localhost', port), ProxyHandler)
    print(f"✅ CORS Proxy running on http://localhost:{port}")
    print(f"📊 Dashboard: http://localhost:{port}/dashboard-v3.html")
    print(f"🔄 Proxying /api/* → {GATEWAY_URL}")
    server.serve_forever()
